components/bsx/Gallery/Item/AvailableActions.vue

```diff
@@ -0,0 +1,221 @@
+import KeyboardEventsMixin from '~/utils/mixins/keyboardEventsMixin'
+import MetaTransactionMixin from '~/utils/mixins/metaMixin'
+import PrefixMixin from '~/utils/mixins/prefixMixin'
+
+const components = {
+  ActionList: () => import('@/components/rmrk/Gallery/Item/ActionList.vue'),
+  AddressInput: () => import('@/components/shared/AddressInput.vue'),
+  BalanceInput: () => import('@/components/shared/BalanceInput.vue'),
+  SubmitButton: () => import('@/components/base/SubmitButton.vue'),
+  Loader: () => import('@/components/shared/Loader.vue'),
+}
+
+@Component({ components })
+export default class AvailableActions extends mixins(
+  PrefixMixin,
+  KeyboardEventsMixin,
+  MetaTransactionMixin,
+  AuthMixin
+) {
+  @Prop(String) public currentOwnerId!: string
+  @Prop() public price!: string
+  @Prop() public nftId!: string
+  @Prop(String) public collectionId!: string
+  @Prop({ type: Array, default: () => [] }) public ipfsHashes!: string[]
+
+  private selectedAction: NFTAction | '' = ''
+  private meta: string | number = ''
+
+  get actions() {
+    return getActionList('bsx', this.isOwner, this.isAvailableToBuy)
+  }
+
+  get isOwner(): boolean {
+    this.$consola.log(
+      '{ currentOwnerId, accountId }',
+      this.currentOwnerId,
+      this.accountId
+    )
+
+    return Boolean(
+      this.currentOwnerId &&
+        this.accountId &&
+        isSameAccount(this.currentOwnerId, this.accountId)
+    )
+  }
+
+  get showSubmit() {
+    return this.selectedAction && (!this.showMeta || this.metaValid)
+  }
+
+  get metaValid() {
+    if (typeof this.meta === 'number') {
+      return this.meta >= 0
+    }
+
+    return this.meta
+  }
+
+  get showMeta() {
+    return actionComponent[this.selectedAction]
+  }
+
+  protected iconType(value: string) {
+    return iconResolver[value]
+  }
+
+  protected handleAction(action: NFTAction) {
+    if (shouldUpdate(action, this.selectedAction)) {
+      this.selectedAction = action
+    } else {
+      this.selectedAction = NFTAction.NONE
+      this.meta = ''
+    }
+  }
+
+  get isAvailableToBuy(): boolean {
+    const { price, accountId } = this
+    return Boolean(accountId && Number(price) > 0)
+  }
+
+  private handleSelect(value: NFTAction) {
+    this.selectedAction = value
+    this.meta = ''
+  }
+
+  get isConsume() {
+    return this.selectedAction === 'CONSUME'
+  }
+
+  protected updateMeta(value: string | number) {
+    this.$consola.log(typeof value, value)
+    this.meta = value
+  }
+
+  protected async checkBuyBeforeSubmit() {
+    const nft = await this.$apollo.query({
+      query: nftById,
+      variables: {
+        id: this.nftId,
+      },
+    })
+
+    const {
+      data: { nFTEntity },
+    } = nft
+    // DEV: nFTEntity.price == 0 is a feature to handle the buy flow
+    if (
+      nFTEntity.currentOwner !== this.currentOwnerId ||
+      nFTEntity.burned ||
+      nFTEntity.price == 0 ||
+      nFTEntity.price !== this.price
+    ) {
+      showNotification(
```

Comment:
for those notifications, maybe we could wapper them into like following and put them into `utils/notification`
```js
const infoNotification = (msg) => showNotification(`[INFO] ${msg}`, notificationTypes.info)
const warnNotification = (msg) => showNotification(`[WARN] ${msg}`, notificationTypes.war)
const dangerNotification = (msg) => showNotification(`[ERR] ${msg}`, notificationTypes.danger)
```

Reply:
added to `notification.ts`